### ISSUE TITLE
get_rdataset with cache is broken (perhaps only py3)

### ISSUE THREAD

**Issue by dlovell** [statsmodels/statsmodels#3831]
Results in internet always being used (and failing if not present)

Issue is that the incorrect method is called on the bytes object: should be `decode` (like in the `else` block just below it)

Below I show the usage of the correct method within `ipdb`

```
In [78]: import os

In [79]: cache_path = '/home/dlovell/statsmodels_data/raw.github.com,vincentarelbundock,Rdatasets,master,csv,datasets,sunspots.csv.zip'

In [80]: os.path.exists(cache_path)
Out[80]: True

In [81]: data = sdu._open_cache(cache_path)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-81-fb74558ddc07> in <module>()
----> 1 data = sdu._open_cache(cache_path)

~/anaconda3/envs/pymc3/lib/python3.6/site-packages/statsmodels/datasets/utils.py in _open_cache(cache_path)
    185         data = zlib.decompress(open(cache_path, 'rb').read())
    186         # return as bytes object encoded in utf-8 for cross-compat of cached
--> 187         data = cPickle.loads(data).encode('utf-8')
    188     else:
    189         data = open(cache_path, 'rb').read().decode('zip')

AttributeError: 'bytes' object has no attribute 'encode'
> /home/dlovell/anaconda3/envs/pymc3/lib/python3.6/site-packages/statsmodels/datasets/utils.py(187)_open_cache()
    185         data = zlib.decompress(open(cache_path, 'rb').read())
    186         # return as bytes object encoded in utf-8 for cross-compat of cached
--> 187         data = cPickle.loads(data).encode('utf-8')
    188     else:
    189         data = open(cache_path, 'rb').read().decode('zip')

ipdb> len(cPickle.loads(data).decode('utf-8'))
79218
```

**Comment by dlovell** [comment]
Actually, _get_data does the decode, the solution is just to remove the call to `encode`

**Comment by josef-pkt** [comment]
temporarily prio-high as reminder to look at it

**Comment by josef-pkt** [comment]
Something is broken, but I don't understand why the current unit test doesn't fail.
If I run the unit test, then it passes. Travis had and internet connection during testing, which I guess it still has, but the unit test doesn't fail. 
There is an `assert_(duncan.from_cache)` and from_cache is True in the unit tests.

When I run an example in the interpreter with a different cache directory, then I get the same error as in the description.

This seems to work with **old** cached files.
checking: we have the cache files in the repo, which I assume were created with python 2
https://github.com/statsmodels/statsmodels/tree/master/statsmodels/datasets/tests

So, maybe the underlying problem is that the cache writing with py3 uses a different format/encoding than with py 2.

**Comment by josef-pkt** [comment]
If I didn't make a mistake in shuffling around cache files:
The proposed fix of removing encode causes an exception when reading old (py2) cache files when I run the unit test.
```
    def _get_data(base_url, dataname, cache, extension="csv"):
        url = base_url + (dataname + ".%s") % extension
        try:
            data, from_cache = _urlopen_cached(url, cache)
        except HTTPError as err:
            if '404' in str(err):
                raise ValueError("Dataset %s was not found." % dataname)
            else:
                raise err

>       data = data.decode('utf-8', 'strict')
E       AttributeError: 'str' object has no attribute 'decode'
```

**Comment by josef-pkt** [comment]
proposed fix in #4499 is to change the cache writing on python 3 and not the cache reading as proposed here.

**Comment by josef-pkt** [comment]
bug fixed in #4499

**Comment by josef-pkt** [comment]
@dlovell Thanks for reporting this.
